deployment.md

```diff
@@ -0,0 +1,162 @@
+- <max_price_markup> - the maximum price markup that the paymaster owner can set.
+
+To deploy FixedOracle: run:
+
+```shell
+cast send $ORACLE_FACTORY_ADDRESS "deployFixedOracle(bytes32 salt, int256 _price)" <salt> 100000000 --private-key $FACTORY_OWNER_PRIVATE_KEY -r $PROVIDER_URL -c <chain_id>
+```
+
+The command will return the address of the deployed oracle, that you should save to `$FIXED_ORACLE_ADDRESS`.
+
+---
+
+To deploy TwapOracle: run:
+
+```shell
+cast send $ORACLE_FACTORY_ADDRESS "deployTwapOracle(bytes32,address,uint32,address)" <salt> $NATIVE_POOL_ADDRESS 61 POOL_NATIVE_TOKEN_ADDRESS --private-key $FACTORY_OWNER_PRIVATE_KEY -r $PROVIDER_URL -c <chain_id>
+```
+
+The command will return the address of the deployed oracle, that you should save to `$TWAP_ORACLE_ADDRESS`.
+
+---
+
+To deploy ERC20PaymasterV6 run:
+
+```shell
+cast send $PAYMASTER_FACTORY_ADDRESS "deployPaymaster(bytes32,uint8,address,address,address,address,uint32,address,uint32,uint32,uint256,uint256)" <salt> 0 $TOKEN_ADDRESS 0x5FF137D4b0FDCD49DcA30c7CF57E578a026d2789 $FIXED_ORACLE_ADDRESS $TWAP_ORACLE_ADDRESS 172800 $PAYMASTER_OWNER_ADDRESS <price_markup> <max_price_markup> 30000 50000 --private-key $OWNER_PRIVATE_KEY -r $PROVIDER_URL -c <chain_id>
+```
+
+The command will return the address of the deployed paymaster, that you should save to `$PAYMASTER_ADDRESS`.
+
+## Changes to Wallet settings
+
+Now you should have the paymaster deployed. The next step would be to change the settings of the Wallet so that paymaster can be used.
+
+For each chain you have deployed the paymaster to, you need to modify the `wallet.json` file in the corresponding `networks/<chain_id>` directory in the `clearsync` repository.
+
+You need to add fee token address as key and paymaster address as a value to `paymasters` mapping:
+
+```json
+{
+  "paymaster": [
+    {
+      "<fee_token_address>": "<paymaster_address>"
+    }
+  ]
+}
+```
+
+Where `<fee_token_address>` is the address of the token that the paymaster will operate with, and `<paymaster_address>` is the address of the deployed ERC20 paymaster.
+
+## Operation
+
+### EntryPoint deposit
+
+The paymaster requires a deposit on the EntryPoint to operate. During UserOp execution, EntryPoint will check the paymaster deposit and deduct the required amount from it.
+
+To check the paymaster deposit, run:
+
+```shell
+cast call 0x5FF137D4b0FDCD49DcA30c7CF57E578a026d2789 "balanceOf(address account)" $PAYMASTER_ADDRESS -r $PROVIDER_URL -c <chain_id>
+```
+
+---
+
+Any account can deposit funds to the paymaster, therefore you can use private key of any account with funds (specify it in `$PRIVATE_KEY`).
+
+An amount to top-up the deposit should be specified in <amount>.
+
+To top-up the paymaster deposit, run:
+
+```shell
+cast send 0x5FF137D4b0FDCD49DcA30c7CF57E578a026d2789 "depositTo(address account)" $PAYMASTER_ADDRESS --value <amount> --private-key $PRIVATE_KEY -r $PROVIDER_URL -c <chain_id>
+```
```

Comment:
Do you think it would be beneficial if there would be two commands – the first example shows how to do that on EntryPoint, the seconds one shows how to do the same but with a paymaster contract method?

Reply:
I think, it would be nice to have only the Paymaster's command. This way, we omit one parameter for the user (the paymaster address), and don't bother them with a choice (what command should they choose).

Reply:
Right, I agree

Reply:
60879dbb5b1bb13b48bb5bc131b6e990d3862a88